xarray/tests/test_backends.py

```diff
@@ -278,6 +278,32 @@ def test_roundtrip_datetime_data(self):
             self.assertEquals(actual.t0.encoding['units'],
                               'days since 1950-01-01')
 
+    def test_roundtrip_netcdftime_datetime_data(self):
+        from .test_coding import _non_standard_netcdftime_date_types
```

Comment:
I had a comment on this earlier and just want to make sure it doesn't get lost; I think we do need to do something about this, otherwise roundtrips using standard calendars outside the Timestamp-valid range could be unreliable.

>If I use `_all_netcdftime_date_types` here (i.e. including standard calendar date types) this test fails, because of an obscure issue related to the fact that the repr of netcdftime objects is not zero padded (e.g. the date 0001-01-01 is printed as 1-01-01).  If one passes this repr to `pd.Timestamp` one gets:
>```
>In [1]: import pandas as pd
>
>In [2]: pd.Timestamp('1-01-01')
>Out[2]: Timestamp('2001-01-01 00:00:00')
>```
>So in this case, the times associated with the coordinate `t` in this Dataset get encoded with a units attribute of `'days since    1-01-01 00:00:00`, but for standard calendars this datestring gets decoded by passing the datestring to a pandas Timestamp object -- in that case it gets (incorrectly) decoded as a date in 2001.  The cleanest fix might be updating the repr upstream in netcdftime to be zero padded.  Does that sound like the right move or should we try to work around it in xarray?  A zero-padded date gives the desired behavior:
>```
>In [3]: pd.Timestamp('0001-01-01')
>---------------------------------------------------------------------------
>OutOfBoundsDatetime                       Traceback (most recent call last)
><ipython-input-8-5ddcf5888def> in <module>()
>----> 1 pd.Timestamp('0001-01-01')
>
>pandas/tslib.pyx in pandas.tslib.Timestamp.__new__ (pandas/tslib.c:10447)()
>
>pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas/tslib.c:27555)()
>
>pandas/tslib.pyx in pandas.tslib.convert_str_to_tsobject (pandas/tslib.c:29955)()
>
>pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas/tslib.c:28700)()
>
>pandas/tslib.pyx in pandas.tslib._check_dts_bounds (pandas/tslib.c:32835)()
>
>OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 1-01-01 00:00:00
>```